elements/pf-tabs/pf-tab.ts

```diff
@@ -69,84 +72,102 @@ import styles from './pf-tab.css';
   private icons!: Array<HTMLElement>;
 
-  @property({ type: Boolean }) manual = false;
-
   @observed
   @property({ reflect: true, type: Boolean }) active = false;
 
   @observed
   @property({ reflect: true, type: Boolean }) disabled = false;
 
-  #internals = this.attachInternals();
+  @consume({ context, subscribe: true })
+  @property({ attribute: false })
+  private ctx?: PfTabsContext;
+
+  #internals = InternalsController.of(this, { role: 'tab' });
 
   override connectedCallback() {
     super.connectedCallback();
     this.id ||= getRandomId(this.localName);
-    this.#internals.role = 'tab';
-    this.#internals.ariaDisabled = this.#setInternalsAriaDisabled();
     this.addEventListener('click', this.#onClick);
+    this.addEventListener('keydown', this.#onKeydown);
     this.addEventListener('focus', this.#onFocus);
   }
 
+  override willUpdate() {
+    const { borderBottom, box, fill, manual, vertical } = this.ctx ?? {};
+    this.toggleAttribute('fill', fill);
+    this.toggleAttribute('manual', manual);
+    this.toggleAttribute('vertical', vertical);
+    if (box) {
+      this.setAttribute('box', box);
+    } else {
+      this.removeAttribute('box');
+    }
+    if (borderBottom) {
+      this.setAttribute('border-bottom', borderBottom);
+    } else {
+      this.removeAttribute('border-bottom');
+    }
+  }
+
   render() {
+    const { active } = this;
+    const { box, fill = false, vertical = false } = this.ctx ?? {};
+    const light = box === 'light';
+    const dark = box === 'dark';
     return html`
-      <button part="button" ?disabled="${this.disabled}">
+      <div id="button"
+           part="button"
+           class="${classMap({ active, box: !!box, dark, light, fill, vertical })}">
         <slot name="icon"
               part="icon"
               ?hidden="${!this.icons.length}"
               @slotchange="${() => this.requestUpdate()}"></slot>
         <slot part="text"></slot>
-      </button>
+      </div>
     `;
   }
 
   #onClick() {
-    if (this.disabled || this.#internals.ariaDisabled === 'true') {
-      return;
+    if (!this.disabled) {
+      this.#activate();
+    }
+  }
+
+  #onKeydown(event: KeyboardEvent) {
+    if (!this.disabled) {
+      switch (event.key) {
+        case 'Enter': this.#activate();
+      }
     }
-    this.#activate();
-    this.button.focus();
   }
 
   #onFocus() {
-    if (this.manual || this.#internals.ariaDisabled === 'true') {
-      return;
+    if (!this.ctx?.manual && !this.disabled) {
+      this.#activate();
     }
-    this.#activate();
   }
 
   #activate() {
-    this.active = true;
+    return this.dispatchEvent(new TabExpandEvent(this));
   }
 
-  private _activeChanged(oldVal: boolean, newVal: boolean) {
-    if (oldVal !== newVal && newVal === true) {
-      this.dispatchEvent(new TabExpandEvent(this));
+  private _activeChanged(old: boolean) {
+    if (this.active && !old) {
+      this.#activate();
```

Comment:
I believe we need to set the `#internals.ariaSelected` state here as well, but only if its not disabled.

```Suggestion
  private _activeChanged(old: boolean) {
    if (this.active && !old) {
      this.#activate();
    }
    if (this.active && !this.disabled) {
      this.#internals.ariaSelected = 'true';
    } else {
      this.#internals.ariaSelected = 'false';
    }
```

Reply:
@nikkimk I'm eager to hear your input on this

Reply:
Another thing to note on this same code chunk is looks like we are no longer logging a warning about attempted setting of `disabled` tabs to active.  Oversight or is there reasoning for removal?

![image](https://github.com/patternfly/patternfly-elements/assets/13666/89510a3b-6839-46cd-869e-2b399bbe69e5)

Reply:
49f7ec92

Reply:
c8b39e60 cc @nikkimk